web/client/utils/FilterUtils.js

```diff
@@ -834,21 +834,57 @@ export const cqlArrayField = function(attribute, operator, value) {
     }
 };
 
+/**
+ * Process CQL filter value, properly converts wildcards at the start and end of the string
+ * so that "like" and "ilike" condition will get "*text" or "text*" conditions as "%text" and "text%".
+ * "like" and "ilike" operators value will be turned into "%value%" if no wildcards were used.
+ * All other operators will receive value as-is, no wildcard processing is applied
+ * @param value
+ * @param operator
+ * @returns {string}
+ */
+export const processCqlWildcards = function(value, operator) {
+    const escapedQuotes = escapeCQLStrings(value);
+    const startCharIsWildcard = value.startsWith('*');
+    const endCharIsWildcard = value.endsWith('*');
+    const startsWithWildcard = startCharIsWildcard && !endCharIsWildcard;
```

Comment:
Starts or ends was only examples. Like filter can contain * everywhere.
We need to follow the same convention in the tooltip:

> use * for any number of any char
> use . for a single char
> use ! to escape above two (* and .) 

WFS requests in fact are sent using this convention (Try to use the advanced filter in feature grid):
```
<ogc:PropertyIsLike matchCase="false" wildCard="*" singleChar="." escapeChar="!">
```
The filter, when no wildcard is used is generated like that (inserting `| STATE_NAME | ilike | a |`)

```
<ogc:PropertyName>STATE_NAME</ogc:PropertyName><ogc:Literal>*a*</ogc:Literal>
```
![image](https://user-images.githubusercontent.com/1279510/154443641-39bd505c-e973-4815-822c-0d8fe4ce7662.png)

Accordingly with the [reference](https://docs.geoserver.org/latest/en/user/styling/ysld/reference/filters.html) CQL uses `%` for multi-character and _ for single char. 

So we need to:

- [ ]  produce a CQL_FILTERS that transforms the string of the operators like and ilike like that: 
    - convert `*` and `.` with `%` and `_`
    - not convert when `!.` or `!*`
    - Escapes when `%` or `_" are present, ( Don't remember the escape char, I think `\.` and `\%` will work, but need to test)

Please document this accordingly in this method.

- [ ] I note right now that inserting `| STATE_NAME | ilike | a *|` in advanced filter of feature grid we have a filter like 
```
<ogc:PropertyName>STATE_NAME</ogc:PropertyName><ogc:Literal>*a**</ogc:Literal>
```

That inconsitent with what we declared in the issue. Therefore we should see to fix also this behavior.

Reply:
I think I misunderstood requirements, sorry.
All cases listed above should be covered with the changes applied by the last commit.